Failed sim test ci

**Describe the bug**

Sim tests randomly failed with

```
Simulation environment "multi-fork" is stopping: Sim run timedout in 437000 ms 
DockerRunner stopping 'node-1-el-geth'...
DockerRunner stopping 'node-2-el-nethermind'...
DockerRunner stopping 'node-3-el-geth'...
DockerRunner stopping 'node-4-el-nethermind'...
file:///home/runner/work/lodestar/lodestar/packages/utils/src/sleep.ts:24
      reject(new ErrorAborted());
             ^
ErrorAborted: Aborted 
    at EventTarget.onAbort (file:///home/runner/work/lodestar/lodestar/packages/utils/src/sleep.ts:24:14)
    at EventTarget.[nodejs.internal.kHybridDispatch] (node:internal/event_target:731:20)
    at EventTarget.dispatchEvent (node:internal/event_target:673:26)
    at abortSignal (node:internal/abort_controller:308:10)
    at AbortController.abort (node:internal/abort_controller:338:5)
    at SimulationEnvironment.stop (file:///home/runner/work/lodestar/lodestar/packages/cli/test/utils/simulation/SimulationEnvironment.ts:237:29)
    at Timeout._onTimeout (file:///home/runner/work/lodestar/lodestar/packages/cli/test/utils/simulation/SimulationEnvironment.ts:145:12)
    at listOnTimeout (node:internal/timers:564:17)
    at processTimers (node:internal/timers:507:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

for example https://github.com/ChainSafe/lodestar/actions/runs/3792403436/jobs/6448569972 https://github.com/ChainSafe/lodestar/actions/runs/3798818524/jobs/6471882574

**Expected behavior**

No error

--- a/packages/cli/test/utils/simulation/SimulationEnvironment.ts
+++ b/packages/cli/test/utils/simulation/SimulationEnvironment.ts
@@ -141,15 +141,27 @@ export class SimulationEnvironment {
   }
 
   async start(opts: StartOpts): Promise<void> {
+    const currentTime = Date.now();
     setTimeout(() => {
-      this.stop(1, `Sim run timedout in ${opts.runTimeoutMs} ms `).catch((e) => console.error("Error on stop", e));
+      const slots = this.clock.getSlotFor(currentTime + opts.runTimeoutMs);
+      const epoch = this.clock.getEpochForSlot(slots);
+      const slot = this.clock.getSlotIndexInEpoch(slots);
+
+      this.stop(1, `Sim run timedout in ${opts.runTimeoutMs}ms (approx. ${epoch}/${slot}).`).catch((e) =>
+        console.error("Error on stop", e)
+      );
     }, opts.runTimeoutMs);
 
     const msToGenesis = this.clock.msToGenesis();
     const startTimeout = setTimeout(() => {
-      this.stop(1, `Start sequence not completed before genesis, in ${msToGenesis} ms`).catch((e) =>
-        console.error("Error on stop", e)
-      );
+      const slots = this.clock.getSlotFor(currentTime + msToGenesis);
+      const epoch = this.clock.getEpochForSlot(slots);
+      const slot = this.clock.getSlotIndexInEpoch(slots);
+
+      this.stop(
+        1,
+        `Start sequence not completed before genesis, in ${msToGenesis}ms (approx. ${epoch}/${slot}).`
+      ).catch((e) => console.error("Error on stop", e));
     }, msToGenesis);
 
     try {

--- a/packages/cli/test/utils/simulation/TableReporter.ts
+++ b/packages/cli/test/utils/simulation/TableReporter.ts
@@ -1,9 +1,10 @@
 /* eslint-disable no-console */
 import {Slot} from "@lodestar/types";
+import {HeadSummary} from "./assertions/defaults/headAssertion.js";
 import {defaultAssertions} from "./assertions/defaults/index.js";
 import {SimulationReporter} from "./interfaces.js";
 import {TableRenderer} from "./TableRenderer.js";
-import {arrayGroupBy, avg} from "./utils/index.js";
+import {arrayGroupBy, avg, isUnique} from "./utils/index.js";
 
 export class TableReporter extends SimulationReporter<typeof defaultAssertions> {
   private lastPrintedSlot = -1;
@@ -69,22 +70,22 @@ export class TableReporter extends SimulationReporter<typeof defaultAssertions> 
       }
     }
 
-    const finalizedSlots = nodes.map((node) => stores["finalized"][node.cl.id][slot] ?? "-");
-    const finalizedSlotsUnique = new Set(finalizedSlots);
+    const finalizedSlots: number[] = [];
+    const inclusionDelay: number[] = [];
+    const attestationCount: number[] = [];
+    const heads: HeadSummary[] = [];
+    const peerCount: number[] = [];
 
-    const inclusionDelay = nodes.map((node) => stores["inclusionDelay"][node.cl.id][slot] ?? "-");
-    const inclusionDelayUnique = new Set(inclusionDelay);
+    for (const node of nodes) {
+      finalizedSlots.push(stores["finalized"][node.cl.id][slot] ?? "-");
+      inclusionDelay.push(stores["inclusionDelay"][node.cl.id][slot] ?? "-");
+      attestationCount.push(stores["attestationsCount"][node.cl.id][slot] ?? "-");
+      heads.push(stores["head"][node.cl.id][slot] ?? "-");
+      peerCount.push(stores["connectedPeerCount"][node.cl.id][slot] ?? "-");
+    }
 
-    const attestationCount = nodes.map((node) => stores["attestationsCount"][node.cl.id][slot] ?? "-");
-    const attestationCountUnique = new Set(attestationCount);
-
-    const heads = nodes.map((node) => stores["head"][node.cl.id][slot]);
     const head0 = heads.length > 0 ? heads[0] : null;
     const nodesHaveSameHead = heads.every((head) => head?.blockRoot !== head0?.blockRoot);
-
-    const peerCount = nodes.map((node) => stores["connectedPeerCount"][node.cl.id][slot] ?? "-");
-    const peerCountUnique = new Set(peerCount);
-
     const errorCount = errors.filter((e) => e.slot === slot).length;
 
     this.table.addRow({
@@ -92,10 +93,10 @@ export class TableReporter extends SimulationReporter<typeof defaultAssertions> 
       eph: epochStr,
       slot: head0 ? head0.slot : "-",
       head: head0 ? (nodesHaveSameHead ? `${head0?.blockRoot.slice(0, 6)}..` : "different") : "-",
-      finzed: finalizedSlotsUnique.size === 1 ? finalizedSlots[0] : finalizedSlots.join(","),
-      peers: peerCountUnique.size === 1 ? peerCount[0] : peerCount.join(","),
-      attCount: attestationCountUnique.size === 1 ? attestationCount[0] : "---",
-      incDelay: inclusionDelayUnique.size === 1 ? inclusionDelay[0].toFixed(2) : "---",
+      finzed: isUnique(finalizedSlots) ? finalizedSlots[0] : finalizedSlots.join(","),
+      peers: isUnique(peerCount) ? peerCount[0] : peerCount.join(","),
+      attCount: isUnique(attestationCount) ? attestationCount[0] : "---",
+      incDelay: isUnique(inclusionDelay) ? inclusionDelay[0].toFixed(2) : "---",
       errors: errorCount,
     });
   }

--- a/packages/cli/test/utils/simulation/TableReporter.ts
+++ b/packages/cli/test/utils/simulation/TableReporter.ts
@@ -44,17 +44,14 @@ export class TableReporter extends SimulationReporter<typeof defaultAssertions> 
       if (epoch - 1 < forkConfig.ALTAIR_FORK_EPOCH) {
         this.table.addEmptyRow("Att Participation: N/A - SC Participation: N/A");
       } else {
-        // attestationParticipation is calculated at first slot of an epoch
-        const participation = nodes.map((node) => stores["attestationParticipation"][node.cl.id][slot] ?? 0);
-        const head = avg(participation.map((p) => p.head)).toFixed(2);
-        const source = avg(participation.map((p) => p.source)).toFixed(2);
-        const target = avg(participation.map((p) => p.target)).toFixed(2);
-
         // As it's printed on the first slot of epoch we need to get the previous epoch
         const startSlot = clock.getFirstSlotOfEpoch(epoch - 1);
         const endSlot = clock.getLastSlotOfEpoch(epoch - 1);
         const nodesSyncParticipationAvg: number[] = [];
+        const participation: {head: number; source: number; target: number}[] = [];
+
         for (const node of nodes) {
+          participation.push(stores["attestationParticipation"][node.cl.id][slot] ?? 0);
           const syncCommitteeParticipation: number[] = [];
           for (let slot = startSlot; slot <= endSlot; slot++) {
             syncCommitteeParticipation.push(stores["syncCommitteeParticipation"][node.cl.id][slot]);
@@ -62,6 +59,10 @@ export class TableReporter extends SimulationReporter<typeof defaultAssertions> 
           nodesSyncParticipationAvg.push(avg(syncCommitteeParticipation));
         }
 
+        // attestationParticipation is calculated at first slot of an epoch
+        const head = avg(participation.map((p) => p.head)).toFixed(2);
+        const source = avg(participation.map((p) => p.source)).toFixed(2);
+        const target = avg(participation.map((p) => p.target)).toFixed(2);
         const syncParticipation = avg(nodesSyncParticipationAvg).toFixed(2);
 
         this.table.addEmptyRow(

--- a/packages/cli/test/utils/simulation/TableReporter.ts
+++ b/packages/cli/test/utils/simulation/TableReporter.ts
@@ -4,7 +4,7 @@ import {HeadSummary} from "./assertions/defaults/headAssertion.js";
 import {defaultAssertions} from "./assertions/defaults/index.js";
 import {SimulationReporter} from "./interfaces.js";
 import {TableRenderer} from "./TableRenderer.js";
-import {arrayGroupBy, avg, isUnique} from "./utils/index.js";
+import {arrayGroupBy, avg, arrayIsUnique} from "./utils/index.js";
 
 export class TableReporter extends SimulationReporter<typeof defaultAssertions> {
   private lastPrintedSlot = -1;
@@ -94,10 +94,10 @@ export class TableReporter extends SimulationReporter<typeof defaultAssertions> 
       eph: epochStr,
       slot: head0 ? head0.slot : "-",
       head: head0 ? (nodesHaveSameHead ? `${head0?.blockRoot.slice(0, 6)}..` : "different") : "-",
-      finzed: isUnique(finalizedSlots) ? finalizedSlots[0] : finalizedSlots.join(","),
-      peers: isUnique(peerCount) ? peerCount[0] : peerCount.join(","),
-      attCount: isUnique(attestationCount) ? attestationCount[0] : "---",
-      incDelay: isUnique(inclusionDelay) ? inclusionDelay[0].toFixed(2) : "---",
+      finzed: !arrayIsUnique(finalizedSlots) ? finalizedSlots[0] : finalizedSlots.join(","),
+      peers: !arrayIsUnique(peerCount) ? peerCount[0] : peerCount.join(","),
+      attCount: !arrayIsUnique(attestationCount) ? attestationCount[0] : "---",
+      incDelay: !arrayIsUnique(inclusionDelay) ? inclusionDelay[0].toFixed(2) : "---",
       errors: errorCount,
     });
   }

--- a/packages/cli/test/utils/simulation/TableReporter.ts
+++ b/packages/cli/test/utils/simulation/TableReporter.ts
@@ -86,7 +86,7 @@ export class TableReporter extends SimulationReporter<typeof defaultAssertions> 
     }
 
     const head0 = heads.length > 0 ? heads[0] : null;
-    const nodesHaveSameHead = heads.every((head) => head?.blockRoot !== head0?.blockRoot);
+    const nodesHaveSameHead = heads.every((head) => head?.blockRoot === head0?.blockRoot);
     const errorCount = errors.filter((e) => e.slot === slot).length;
 
     this.table.addRow({

--- a/packages/cli/test/utils/simulation/utils/index.ts
+++ b/packages/cli/test/utils/simulation/utils/index.ts
@@ -67,3 +67,6 @@ export const arrayGroupBy = <T>(
 export function strFixedSize(str: string, width: number): string {
   return str.padEnd(width).slice(0, width);
 }
+
+export const isUnique = <T>(arr: T[], predicate?: (val: T) => unknown): boolean =>
+  arr.length === new Set(predicate ? arr.map(predicate) : arr).size;

--- a/packages/cli/test/utils/simulation/utils/index.ts
+++ b/packages/cli/test/utils/simulation/utils/index.ts
@@ -68,5 +68,5 @@ export function strFixedSize(str: string, width: number): string {
   return str.padEnd(width).slice(0, width);
 }
 
-export const isUnique = <T>(arr: T[], predicate?: (val: T) => unknown): boolean =>
+export const arrayIsUnique = <T>(arr: T[], predicate?: (val: T) => unknown): boolean =>
   arr.length === new Set(predicate ? arr.map(predicate) : arr).size;

